frontend/src/concepts/areas/utils.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  DashboardConfigKind,
+  DashboardCommonConfig,
+  DataScienceClusterKindStatus,
+} from '~/k8sTypes';
+import { SupportedArea } from './types';
+import { SupportedAreasStateMap } from './const';
+
+type Output = { [flag: string]: boolean };
+const getFlags = (dashboardConfig: DashboardConfigKind): Output => {
+  const flags = dashboardConfig.spec.dashboardConfig;
+
+  return {
+    ...Object.keys(flags).reduce<Output>((flagState, key) => {
+      const value: boolean | string | undefined = flags[key as keyof DashboardCommonConfig];
+      if (typeof value === 'boolean') {
+        flagState[key] = key.startsWith('disable') ? !value : value;
+      }
+      return flagState;
+    }, {}),
+    notebookController: dashboardConfig.spec.notebookController?.enabled ?? false,
+  };
+};
+
+export const isAreaAvailable = (
+  area: SupportedArea,
+  dashboardConfig: DashboardConfigKind,
+  dscStatus: DataScienceClusterKindStatus | null,
+): boolean => {
+  const { featureFlags, requiredComponents, reliantAreas } = SupportedAreasStateMap[area];
+
+  const hasReliantAreas = reliantAreas
+    ? reliantAreas.some((area) => isAreaAvailable(area, dashboardConfig, dscStatus))
+    : true;
+
+  if (!hasReliantAreas) {
+    return false;
+  }
+
+  const flagState = getFlags(dashboardConfig);
+  const hasFeatureFlagsNeeds: boolean = featureFlags
```

Comment:
You can infer the type here and similarly below.
```suggestion
  const hasFeatureFlagsNeeds = featureFlags
```